src/Pulp/Browserify.purs

```diff
@@ -27,16 +27,20 @@ import Pulp.Exec (pursBundle)
 import Pulp.Files (outputModules)
 import Pulp.Outputter (getOutputter)
 import Pulp.Project (Project(..))
-import Pulp.Run (makeEntry, jsEscape)
+import Pulp.Run (jsEscape, makeCjsEntry)
 import Pulp.Sorcery (sorcery)
 import Pulp.System.FFI (Callback, runNode)
 import Pulp.System.Files (openTemp)
 import Pulp.System.Stream (WritableStream)
+import Pulp.Validate (failIfUsingEsModulesPsVersion)
 
 action :: Action
 action = Action \args -> do
   out <- getOutputter args
 
+  failIfUsingEsModulesPsVersion out $ Just
+    "Code path reason: browserify only works on CommonJS modules"
```

Comment:
These error messages read a bit strangely to me. What is the significance of `Code path reason:`? Could we instead say something like:

```suggestion
    "Browserify can only be used with CommonJS modules, which were emitted by PureScript compilers pre-v0.15, but your project is using ES modules."
```

Reply:
My goal here was to define a single generic message [via `failIfUsingEsModulesPsVersion`](https://github.com/purescript-contrib/pulp/pull/401#pullrequestreview-905058318) that centralized a lot of info (i.e. "this is no longer supported. See the migration guide via the link") and then provide a better explanation for what caused the error in that particular case. Since the generic message says "This code path..." I thought "Code path reason: X" would clarify that part a bit.

For example, the entire message shown here would be.
```
This code path implicitly uses `purs bundle` or CommonsJS modules, both of which are no longer supported in PureScript v0.15.0. You are using PureScript v0.14.0. See https://github.com/purescript/documentation/blob/master/migration-guides/0.15-Migration-Guide.md
Code path reason: browserify only works on CommonJS modules
```

Reply:
OK, I see. Works for me!